sigh/lib/assets/resign.sh

```diff
@@ -734,23 +745,32 @@ function resign {
             "com.apple.developer.default-data-protection" \
             "com.apple.developer.healthkit" \
             "com.apple.developer.homekit" \
-            "com.apple.developer.icloud-container-environment" \
-            "com.apple.developer.icloud-container-identifiers" \
+            "com.apple.developer.icloud-container-environment|ICLOUD_ENV" \
             "com.apple.developer.icloud-services" \
-            "com.apple.developer.in-app-payments" \
             "com.apple.developer.networking.HotspotConfiguration" \
             "com.apple.developer.networking.multipath" \
             "com.apple.developer.networking.networkextension" \
             "com.apple.developer.networking.vpn.api" \
             "com.apple.developer.nfc.readersession.formats" \
-            "com.apple.developer.pass-type-identifiers|TEAM_ID" \
             "com.apple.developer.siri" \
-            "com.apple.developer.ubiquity-container-identifiers" \
             "com.apple.developer.ubiquity-kvstore-identifier|TEAM_ID" \
             "com.apple.external-accessory.wireless-configuration" \
-            "com.apple.security.application-groups" \
             "inter-app-audio" \
-            "keychain-access-groups|APP_ID")
+            "keychain-access-groups|APP_ID" \
+        )
+
+        # If we change team while resigning, we have no other choice than to use the following entitlements from the PP instead of the App
+        # because they are based on unique identifiers (defined in the developer portal) that can't be shared between teams
+        # If we don't change team while resigning, we should use the following entitlements from the existing App and not from the PP
+        if [[ "$OLD_TEAM_ID" == "$NEW_TEAM_ID" ]]; then
+            ENTITLEMENTS_TRANSFER_RULES+=(\
```

Comment:
Can we have one source array for both `ENTITLEMENTS_TRANSFER_RULES += ` and `BLACKLISTED_KEYS+=` above? I want to reduce the risk someone will accidentally modify one array in the future and forget to modify the other.

Reply:
I was willing to do that but:
- the `ENTITLEMENTS_TRANSFER_RULES` array can include patching rules (`TEAM_ID`, `APP_ID`, and now `ICLOUD_ENV`) that should then be ignored for blacklisting purposes
- we could have near-future cases where a key is not in the blacklist but still added to the transfer rules to be retrieved from the existing app if defined (used as a preferred value)

That mean we would have to maintain a single array with "behavior types" in addition to the potential patching rule(s) for each of the keys.

This should definitely be investigated (along with a whitelist approach), but this goes beyond this "quick fix".